diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/EmptyBlock.kt

```diff
@@ -0,0 +1,103 @@
+
+    companion object {
+        private const val emptyBlockSize = 4
+    }
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var fileName: String? = null
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        fileName = params.fileName
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val configuration = EmptyBlockStyleConfiguration(
+                configRules.getRuleConfig(BRACES_BLOCK_STRUCTURE_ERROR)?.configuration ?: mapOf()
+        )
+        searchNode(node, configuration)
+    }
+
+    private fun searchNode(node: ASTNode, configuration: EmptyBlockStyleConfiguration) {
+        val newNode = when (node.elementType) {
+            THEN -> node.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
+            ELSE -> node.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
+            WHEN -> node.findChildByType(LBRACE)!!.treeNext.treeParent
+            FOR, WHILE, DO_WHILE -> node.findChildByType(BODY)?.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
+            CLASS, OBJECT_DECLARATION -> node.findChildByType(CLASS_BODY)!!.findChildByType(LBRACE)!!.treeNext.treeParent
+            else -> if (node.hasChildOfType(BLOCK))
+                node.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
+            else
+                null
+        } ?: return
+        checkEmptyBlock(newNode, configuration)
+    }
+
+    private fun checkEmptyBlock(node: ASTNode, configuration: EmptyBlockStyleConfiguration) {
+        val children = node.getChildren(null)
+        val isOverride = node.treeParent.findChildByType(MODIFIER_LIST)?.findChildByType(OVERRIDE_KEYWORD)
+        if (children.size < emptyBlockSize) {
+            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "there can't be empty blocks in multi blocks",
+                    node.startOffset) {}
+            if (isOverride != null)
+                return
+            val space = node.findChildByType(RBRACE)!!.treePrev
+            if (configuration.emptyBlockNewline && !space.text.contains("\n")) {
+                BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "different style for empty block",
+                        node.startOffset) {
+                    if (space.elementType == WHITE_SPACE)
+                        (space.treeNext as LeafPsiElement).replaceWithText("\n")
+                    else
+                        node.addChild(PsiWhiteSpaceImpl("\n"), space.treeNext)
```

Comment:
I think this construction already deserves to be extracted into separate utility method, like `leaveOrAppendOneNewline`. If you do so, please also use this new method in your previous rule.